app/src/main/java/com/odysee/app/GoLiveActivity.java

```diff
@@ -0,0 +1,600 @@
+            public void onSuccess(Claim claimResult) {
+                signAndSetupStream();
+                progress.setVisibility(View.GONE);
+                buttonToggleStreaming.setEnabled(true);
+            }
+
+            @Override
+            public void onError(Exception error) {
+                if (error != null) {
+                    showError(error.getMessage());
+                }
+                progress.setVisibility(View.GONE);
+                livestreamControlsView.setVisibility(View.GONE);
+                showPrecheckError(getString(R.string.error_publishing_livestream));
+            }
+        });
+        task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+    /**
+     * Checks if it's possible to stream on {@code channel}, and update the
+     * error text with the reason if not.
+     *
+     * @param channel The channel to check
+     * @return True if it's possible to stream on the channel, false otherwise
+     */
+    private boolean checkCanStreamOnChannel(Claim channel) {
+        if (channel == null || channel.isPlaceholder() || channel.isPlaceholderAnonymous()) {
+            return false;
+        }
+
+        if (channel.getConfirmations() < 1) {
+            textChannelError.setText(getString(R.string.channel_error_pending));
+            textChannelError.setVisibility(View.VISIBLE);
+            return false;
+        }
+
+        // Disabled due to lack of server side checking for now
+//        double effectiveAmount = Double.parseDouble(channel.getMeta().getEffectiveAmount());
+//        if (effectiveAmount < MIN_STREAM_STAKE) {
+//            textChannelError.setText(getString(R.string.channel_error_need_minimum_credits,
+//                    MIN_STREAM_STAKE, channel.getName()));
+//            textChannelError.setVisibility(View.VISIBLE);
+//            return false;
+//        }
+
+        textChannelError.setVisibility(View.GONE);
+        return true;
+    }
+
+    private void showLivestreamingOptions() {
+        progress.setVisibility(View.GONE);
+        livestreamPrecheckView.setVisibility(View.GONE);
+        livestreamOptionsView.setVisibility(View.VISIBLE);
+
+        checkCanStreamOnChannel((Claim) selectChannelSpinner.getSelectedItem());
+    }
+
+    private void updateChannelList(List<Claim> channels) {
+        selectChannelSpinner.setAdapter(new InlineChannelSpinnerAdapter(this, R.layout.spinner_item_channel, channels));
+
+        if (channels.size() > 0) {
+            showLivestreamingOptions();
+        } else {
+            progress.setVisibility(View.GONE);
+            showPrecheckError(getString(R.string.precheck_need_channel));
+        }
+    }
+
+    private void fetchChannels() {
+        if (Lbry.ownChannels != null && Lbry.ownChannels.size() > 0) {
+            updateChannelList(Lbry.ownChannels);
+            return;
+        }
+
+        ClaimListTask task = new ClaimListTask(Claim.TYPE_CHANNEL, null, new ClaimListResultHandler() {
```

Comment:
Use ClaimList supplier for API Level 24+

Reply:
I noticed that, but there's quite a lot of duplicated code by doing that, right?

What's the difference between the two?

Reply:
ClaimListTask is an AsyncTask, which is deprecated. New code should not be using AsyncTask. ClaimListTask remains on repo because there is still code using it.

You could just use the supplier forAPI Level 24+, leaving the AsyncTask to older API Levels.

Reply:
Ok, done in https://github.com/OdyseeTeam/odysee-android/pull/167/commits/b219673e1bc5a589484818151d3202b58793b23a